### ISSUE TITLE
Slower than aiohttp

### ISSUE THREAD

**Issue by XazkerBoy** [sonic182/aiosonic#199]
Hello!
I've tried your library and it worked perfect, until I used the `follow=True` parameter

Here is the test sketch:
```python
import grequests as gr
import asyncio
from aiosonic import HTTPClient
from time import perf_counter
from json import dumps, loads
from aiohttp import ClientSession

class Response:
	def __init__(self, status, headers, content=None):
		self.status_code = status
		self.content = content
		self.text = None
		if self.content is not None:
			self.text = self.content.decode("latin1")
		self.headers = {}
		for h in headers.keys():
			val = headers[h]
			self.headers[h] = val

	def json(self):
		return loads(self.text)

	def __repr__(self):
		return f"<Response [{self.status_code}]>"	# Isn't really needed in this example, just makes responses more comfortable to view

async def request(client, url, include_content):	# AIOSonic request
	resp = await client.get(url, follow=True)	# Follow makes everything much slower
	content = None
	if include_content:
		content = await resp.content()
	resp = Response(resp.status_code, resp.headers, content)
	return resp

async def request_aiohttp(sess, url, include_content):	# AIOHttp request
	async with sess.get(url) as resp:
		content = None
		if include_content:
			content = await resp.read()
		final = Response(resp.status, resp.headers, content)
		return final

async def main(urls, include_content=True, core="aiosonic"):	# Main async method
	client = HTTPClient()	# Client for aiosonic
	sess = ClientSession()	# Client for aiohttp
	resp = []
	for url in urls:
		if core == "aiosonic":
			data = await request(client, url, include_content)
		else:
			data = await request_aiohttp(sess, url, include_content)
		if not data:
			resp.append(None)
			continue
		resp.append(data)
	await client.shutdown()	# Close both clients
	await sess.close()
	return resp

website = "https://google.com"

loop = asyncio.get_event_loop()

fut = asyncio.gather(main([website for _ in range(10)]))	# AIOSonic requests
start = perf_counter()
resp = loop.run_until_complete(fut)
print("AIOSonic:", perf_counter() - start)

fut = asyncio.gather(main([website for _ in range(10)], core="aiohttp"))	# AIOHttp
start = perf_counter()
resp = loop.run_until_complete(fut)
print("AIOHttp:", perf_counter() - start)

start = perf_counter()
reqs = [gr.get(website) for _ in range(10)]
resp = gr.map(reqs)
print("GRequests:", perf_counter() - start)	# GRequests (gevent) [pip install grequests]
```

My results:
```
AIOSonic: 0.7438048629992409
AIOHttp: 0.6540690379988519
GRequests: 0.5792230990009557
```
on a linux server. Maybe it is not fair to compare it with gevent-based one, but it is even slower than aiohttp! Am I doing something wrong or doing redirects requires so much time? Thanks!

**Comment by sonic182** [comment]
Mmm you're right, I maybe is a improvement I have to do at socket reading. The grequests  part is not reading the response content but it still is way faster, nice.

**Comment by sonic182** [comment]
@XazkerBoy Profiling the code, the bottleneck is in ssl communications, aiohttp uses a lot of cython that may be faster for that case. If you test with another ssl website, sometimes it get aiosonic faster, and sometimes aiohttp (I tested with my portfolio www.mogollon.com.ve/en/).

Closing this for now, aiosonic is faster at local servers. Maybe I'll create an issue to move some code to cython on a future
